org.adempiere.base/src/org/compiere/model/MBankStatementLine.java

```diff
@@ -269,5 +278,19 @@ protected boolean updateHeader()
 		}
 		return true;
 	}	//	updateHeader
+
+
+	/**
+	 * Test if the line's date is on the same period as header's, only if BANK_STATEMENT_POST_WITH_DATE_FROM_LINE SysConfig is set to Y 
+	 */
+	public boolean checkDateIfUsedForPosting() {

```

Comment:
@nmicoud - the patch looks fine (I still haven't tested), but I find difficult to understand the name and javadoc of this method.

If I'm reading right it is returning true when the header/line periods match, and when is not validated.
And it returns false when the periods don't match.

If that's the case would be better to explain that in the javadoc, and maybe the name of the method would be better isValidPeriodInLine (or something like that)

I mean, if the method returns a boolean is better to name it using the "is" at the beginning.

"check" is a verb that doesn't tell what's happening inside, this name is probably used when the method inside throws an Exception.

Reply:
hi @CarlosRuiz-globalqss 
you're right, what about isDateConsistentIfUsedForPosting ?
and "If the posting is based on the date of the line (ie SysConfig BANK_STATEMENT_POST_WITH_DATE_FROM_LINE = Y), make sure line and header dates are on the same period" for javadoc ?
I think is important to highlight the fact that it only apply when the SysConfig is activated.
wdyt?

Reply:
Yes- sounds good